modules/tensor_mechanics/src/postprocessors/MaterialTimeStepPostprocessor.C

```diff
@@ -28,38 +28,103 @@ validParams<MaterialTimeStepPostprocessor>()
+
+  params.addRangeCheckedParam<int>("elements_changed_number",
+                                   "elements_changed_number > 0",
+                                   "number of elements used to determine if a change of"
+                                   "time step is required");
+
+  params.addRangeCheckedParam<Real>(
+      "tolerance", "tolerance > 0", "tolerance used to determine if an element has changed");
+
   return params;
 }
 
 MaterialTimeStepPostprocessor::MaterialTimeStepPostprocessor(const InputParameters & parameters)
   : ElementPostprocessor(parameters),
-    _matl_time_step(getMaterialPropertyByName<Real>("matl_timestep_limit"))
+    _use_matl_time_step(getParam<bool>("use_matl_timestep_limit")),
+    _matl_time_step(_use_matl_time_step ? &getMaterialPropertyByName<Real>("matl_timestep_limit")
+                                        : nullptr),
+    _matl_value(std::numeric_limits<Real>::max()),
+    _use_elements_changed(parameters.isParamSetByUser("elements_changed_property")),
+    _changed_property(_use_elements_changed
+                          ? &getMaterialPropertyByName<Real>(
+                                getParam<MaterialPropertyName>("elements_changed_property"))
+                          : nullptr),
+    _changed_property_old(_use_elements_changed
+                              ? &getMaterialPropertyOldByName<Real>(
+                                    getParam<MaterialPropertyName>("elements_changed_property"))
+                              : nullptr),
+    _target(isParamValid("elements_changed_number") ? getParam<int>("elements_changed_number") : 0),
+    _count(0),
+    _tolerance(parameters.isParamSetByUser("tolerance") ? getParam<Real>("tolerance")
+                                                        : TOLERANCE * TOLERANCE),
+    _qp(0)
 {
+  if (_use_elements_changed && !parameters.isParamSetByUser("elements_changed_number"))
+    paramError("elements_changed_number",
+               "needs to be set when elements_changed_property is defined");
+
+  if (!_use_matl_time_step && !_use_elements_changed)
+    mooseError("either use_matl_time_step needs to be true or elements_changed_property defined");
 }
 
 void
 MaterialTimeStepPostprocessor::initialize()
 {
-  _value = std::numeric_limits<Real>::max(); // start w/ the min
+  _matl_value = std::numeric_limits<Real>::max(); // start w/ the min
+  _count = 0;
 }
 
 void
 MaterialTimeStepPostprocessor::execute()
 {
-  for (_qp = 0; _qp < _qrule->n_points(); _qp++)
-    _value = std::min(_value, _matl_time_step[_qp]);
+  if (_use_matl_time_step)
+  {
+    for (_qp = 0; _qp < _qrule->n_points(); _qp++)
+      _matl_value = std::min(_matl_value, (*_matl_time_step)[_qp]);
+  }
+
+  if (_use_elements_changed)
+  {
+    for (_qp = 0; _qp < _qrule->n_points(); ++_qp)
+    {
+      if (!MooseUtils::absoluteFuzzyEqual(
+              (*_changed_property)[_qp], (*_changed_property_old)[_qp], _tolerance))
+      {
+        ++_count;
```

Comment:
This is not really how the capability is described in the documentation because you are counting quadrature points rather than elements. Two options:
1) Leave the code as-is, and update the documentation and command names accordingly.
2) Change it so that you do something like only increment the count a maximum of once per element (and update the documentation to clarify that point).

Reply:
We break the loop (actually return since the function doesn't do anything after that) as soon as we detect one qp that has been changed, so I believe this does count the number of elements. We can update the documentation to indicate that we count elements for which at the property has changed in at least one qp?

Reply:
Oh, I see -- you're right. Yes, just make that a little more clear in the documentation, and I think we'll be good.